From: Shiji Yang <yangshiji66@outlook.com>
Date: Mon, 28 Apr 2025 22:03:18 +0800
Subject: [PATCH] backport: fix unaligned.h header location

unaligned.h was moved from include/asm-generic/ to include/linux/
in 6.12-rc2 kernel.

Link: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=5f60d5f6bbc12e782fac78110b0ee62698f3b576
Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
---
 backport-include/asm/unaligned.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/backport-include/asm/unaligned.h
+++ b/backport-include/asm/unaligned.h
@@ -1,6 +1,10 @@
 #ifndef __BACKPORT_ASM_GENERIC_UNALIGNED_H
 #define __BACKPORT_ASM_GENERIC_UNALIGNED_H
+#if LINUX_VERSION_IS_GEQ(6,12,0)
+#include_next <linux/unaligned.h>
+#else
 #include_next <asm/unaligned.h>
+#endif
 
 #if LINUX_VERSION_IS_LESS(5,7,0)
 static inline u32 __get_unaligned_be24(const u8 *p)
